test/dynamodb.test.js

```diff
@@ -0,0 +1,140 @@
+const test = require('ava');
+const AWS = require('aws-sdk');
+
+const { tableSchema, getConnection, createTables, destroyTables } = require('../src/dynamodb');
+
+test.before(() => {
+  tableSchema([
+    {
+      TableName: 'test-table',
+      AttributeDefinitions: [
+        {
+          AttributeName: 'hashKey',
+          AttributeType: 'S'
+        },
+        {
+          AttributeName: 'rangeKey',
+          AttributeType: 'S'
+        },
+        {
+          AttributeName: 'gsiHashKey',
+          AttributeType: 'S'
+        },
+        {
+          AttributeName: 'lsiRangeKey',
+          AttributeType: 'S'
+        }
+      ],
+      KeySchema: [
+        {
+          AttributeName: 'hashKey',
+          KeyType: 'HASH'
+        },
+        {
+          AttributeName: 'rangeKey',
+          KeyType: 'RANGE'
+        }
+      ],
+      ProvisionedThroughput: {
+        ReadCapacityUnits: 10,
+        WriteCapacityUnits: 10
+      },
+      GlobalSecondaryIndexes: [
+        {
+          IndexName: 'test-gsi',
+          KeySchema: [
+            {
+              AttributeName: 'gsiHashKey',
+              KeyType: 'HASH'
+            },
+            {
+              AttributeName: 'rangeKey',
+              KeyType: 'RANGE'
+            }
+          ],
+          ProvisionedThroughput: {
+            ReadCapacityUnits: 10,
+            WriteCapacityUnits: 10
+          },
+          Projection: {
+            ProjectionType: 'ALL'
+          }
+        }
+      ],
+      LocalSecondaryIndexes: [
+        {
+          IndexName: 'example-lsi',
+          KeySchema: [
+            {
+              AttributeName: 'hashKey',
+              KeyType: 'HASH'
+            },
+            {
+              AttributeName: 'lsiRangeKey',
+              KeyType: 'RANGE'
+            }
+          ],
+          Projection: {
+            ProjectionType: 'ALL'
+          }
+        }
+      ]
+    }
+  ]);
+});
+
+test.after(() => {
+  tableSchema([]);
+});
+
+async function assertTablesPresent (t, client, expected, message) {
```

Comment:
I'm surprised that this behavior isn't covered in the existing dynamodb tests like `test/dynamodb-basic-serial.test.js`.

Can you call out what is unique about these tests?

Reply:
Those tests are indirectly testing the createTable functionality, but not the destroyTable functionality. Even beyond that, it will be more obvious whats going on when these fail. A table will be present/missing when not expected vs a possibly misleading/confusing error message from aws-sdk.

Reply:
You had me at "will be more obvious whats going on when these fail"